integrations/weaviate/src/haystack_integrations/document_stores/weaviate/auth.py

```diff
@@ -0,0 +1,170 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field, fields
+from typing import Any, Dict, Type
+
+from haystack.core.errors import DeserializationError
+from haystack.core.serialization import default_from_dict, default_to_dict
+from haystack.utils.auth import Secret, deserialize_secrets_inplace
+
+from weaviate.auth import AuthApiKey as WeaviateAuthApiKey
+from weaviate.auth import AuthBearerToken as WeaviateAuthBearerToken
+from weaviate.auth import AuthClientCredentials as WeaviateAuthClientCredentials
+from weaviate.auth import AuthClientPassword as WeaviateAuthClientPassword
+
+
+@dataclass
+class AuthCredentials(ABC):
+    """
+    Base class for all auth credentials supported by WeaviateDocumentStore.
+    Can be used to deserialize from dict any of the supported auth credentials.
+    """
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the object to a dictionary representation for serialization.
+        """
+        # We assume all fields are Secret instances
+        _fields = {f.name: getattr(self, f.name).to_dict() for f in fields(self)}
+
+        return default_to_dict(
+            self,
+            **_fields,
+        )
```

Comment:
I'd be more comfortable if we used an explicit enum instead of the full type name. That way, it won't break if we move/rename the underlying classes.